src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -98,7 +98,7 @@ private void AddNullableAnnotations(TypeSymbolWithAnnotations typeOpt)
             }
 
             if (format.MiscellaneousOptions.IncludesOption(SymbolDisplayMiscellaneousOptions.IncludeNullableReferenceTypeModifier) &&
-                !typeOpt.IsNullableType() &&
+                !typeOpt.IsNullableType() && !typeOpt.IsValueType &&
```

Comment:
>!typeOpt.IsValueType && [](start = 45, length = 23)

I didn't understand what's a scenario were we put a nullable annotation on a value type, but it's not `Nullable<T>`. 
Is the when loading bad metadata?

Reply:
> I didn't understand what's a scenario were we put a nullable annotation on a value type, but it's not Nullable<T>. 
Is the when loading bad metadata?

On the contrary, binding from source does that, always did.

---
In reply to: [234741474](https://github.com/dotnet/roslyn/pull/31212#discussion_r234741474) [](ancestors = 234741474)

Reply:
From our discussion, some of the weird combinations (`int` marked as nullable) only come from PE.
You mentioned that one reason for letting such combinations into the compiler is to avoid cycles. Do you remember what's a scenario? 
In particular, I wonder why is that problem unique to nullability, but was ok for decoding dynamic or tuple names?

If it were possible, I still think it would be better for metadata import to ignore, rather than to propagate, bad metadata combinations. I'd expect all value types (except `Nullable<T>`) should be imported as oblivious. I believe that is what existing metadata import tends to do.

Along those lines, why encode a nullable annotation for value types at all? We don't encode a dynamic flag for types that are not object, and we don't encode tuple names for types that are not tuples.

---
In reply to: [234742698](https://github.com/dotnet/roslyn/pull/31212#discussion_r234742698) [](ancestors = 234742698,234741474)